ui-core/src/components/inputs/datePicker/__tests__/useCalendarPicker.spec.ts

```diff
@@ -0,0 +1,168 @@
+import { renderHook, act } from '@testing-library/react';
+import { beforeEach, afterEach, describe, expect, it, vi } from 'vitest';
+import useCalendarPicker, {
+  INVALID_SELECTED_SLOT_ERROR,
+  INVALID_SELECTABLE_SLOT_ERROR,
+  INVALID_SELECTED_SLOT_BASED_ON_SELECTABLE_SLOT_ERROR,
+  INVALID_INITIAL_DATE_ERROR,
+} from '../useCalendarPicker';
+import { generateSequentialDates } from '../utils';
+
+const errorsToIgnore = [
+  INVALID_SELECTED_SLOT_ERROR,
+  INVALID_SELECTABLE_SLOT_ERROR,
+  INVALID_SELECTED_SLOT_BASED_ON_SELECTABLE_SLOT_ERROR,
+  INVALID_INITIAL_DATE_ERROR,
+];
+describe('useCalendarPicker', () => {
+  const january = 0;
+  const february = 1;
+  const june = 5;
+  const july = 6;
+  const august = 7;
+  const december = 11;
+  const onDayClick = vi.fn();
```

Comment:
I moved this outside the `describe`, but I kept 0 for January because these variables are directly used to instantiate `Date` objects, which expect January to be indexed as 0. I stored thes indexex for readability purposes.

Reply:
Got it, thanks :+1: